src/oci/runtime/mod.rs

```diff
@@ -0,0 +1,77 @@
+//! OCI runtime implementations.
+
+use crate::oci::spec::runtime::LinuxResources;
+use anyhow::Result;
+use async_trait::async_trait;
+use strum::{AsRefStr, Display, EnumString, IntoStaticStr};
+use tokio::{process::Command, signal::unix::SignalKind};
+
+pub mod cli;
+
+#[async_trait]
+/// Runtime is the trait for implementing possible interactions with an OCI compatible container.
+pub trait Runtime {
+    type Container: Container + Sized + Send + Sync;
+
+    /// Create a new container.
+    async fn create(&mut self) -> Result<Self::Container>;
+
+    /// Execute the user defined process in a created container.
+    async fn start(&mut self, container: &mut Self::Container) -> Result<()>;
+
+    /// Delete any resources held by the container often used with detached container.
+    async fn delete(&mut self, container: &mut Self::Container) -> Result<()>;
+
+    /// Suspend all processes inside the container.
+    async fn pause(&mut self, container: &mut Self::Container) -> Result<()>;
+
+    /// Resumes all processes that have been previously paused.
+    async fn resume(&mut self, container: &mut Self::Container) -> Result<()>;
+
+    /// Send the specified signal to the container's init process.
+    async fn kill(
+        &mut self,
+        container: &mut Self::Container,
+        signal_kind: SignalKind,
+    ) -> Result<()>;
+
+    /// Update container resource constraints.
+    async fn update(
+        &mut self,
+        container: &mut Self::Container,
+        resources: &LinuxResources,
+    ) -> Result<()>;
+
+    /// Execute the provided process inside the container.
+    async fn exec(&self, container: &Self::Container, command: &Command) -> Result<()>;
+
+    /// Retrieve container resource statistics.
+    async fn stats(&self, container: &Self::Container) -> Result<ContainerStats>;
+
+    /// Retrieve the state of a container.
+    async fn state(&self, container: &Self::Container) -> Result<ContainerState>;
+}
+
+/// Container is the implementation of an OCI compatible container.
+pub trait Container {}
```

Comment:
Hmm, I am not sure why we need this.

Reply:
Removed this trait because `Runtime` is now `Container`.